src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListServiceImpl.java

```diff
@@ -424,6 +426,15 @@ public void addGermplasmListEntriesToAnotherList(final Integer destinationListId
 		this.addGermplasmEntriesModelsToList(destinationGermplasmList, addGermplasmEntriesModels);
 	}
 
+	@Override
+	public GermplasmListGeneratorDTO cloneGermplasmList(final Integer listId, final GermplasmListGeneratorDTO listGeneratorDTO,
+		final Integer loggedInUser) {
+		final GermplasmList destList = this.createGermplasmList(listGeneratorDTO, loggedInUser);
+		this.daoFactory.getGermplasmListDAO().copyGermplasmListEntries(listId, destList.getId(), loggedInUser);
```

Comment:
I think better to have separate queries in respective DAOs for all entitites to be copied
```
this.daoFactory.getGermplasmListDataDAO().copyEntries(sourceListId, destinationListId);
this.daoFactory.getGermplasmListDataViewDAO().copyEntries(sourceListId, destinationListId);
this.daoFactory.getGermplasmListDataDetailDAO().copyEntries(sourceListId, destinationListId. loggedInUser);
```